### PR TITLE
Add windows

### PR DESCRIPTION
**Pull request by ericpre**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This PR add windows builds (cli, gui and python). There are a couple of patches necessary to find libraries on windows and to improve the build of the python binding. I will submit them upstream.

### CODE REVIEW

**Comment by ericpre** [comment]
CI issue should be fixed by https://github.com/conda-forge/conda-smithy/pull/1250. I will do a re-render once a new conda-smithy is available.

**Comment by douglowe** [comment]
The changes mostly look sensible to me - I don't know what the lint'ing error is though, perhaps we should ask on the conda-forge gitter channel, see if someone can advise on this?

I will also download the changes this weekend, and will check them locally.

**Comment by ericpre** [comment]
Thanks @douglowe, if could confirm that the macOS changed are fine, that would be like great.

The app linter warning is related to https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#app-section, maybe the linter simply doesn't know about it when used with outputs?!

**Comment by douglowe** [comment]
hi @ericpre - I've checked the OSX build locally, and everything works alright for me.

Okay, I see what the app information does now. I've checked the channel in Anaconda Navigator on OSX after installing the gui package - I can't see prismatic-gui there, but as the OSX app setup is awkward, maybe that's not surprising. Does it work okay for you on windows?

**Comment by ericpre** [comment]
Thanks for checking the `osx` package!
 
> Okay, I see what the app information does now. I've checked the channel in Anaconda Navigator on OSX after installing the gui package - I can't see prismatic-gui there, but as the OSX app setup is awkward, maybe that's not surprising. Does it work okay for you on windows?

I have exactly the same on windows (it does no appear in the navigator) but that being said, it is not clear to me how it works: for example, it seems that the package doesn't need to be installed to appear in the anaconda navigator, which would imply that this information is pulled from somewhere else. It looks to me that there are limited documentation on this feature, so we will need to ask the right place and the right people!

In this PR, I did this here because I was setting up the shortcut in the start menu on windows and I followed what I did for hyperspyui, for which the "app" section is working fine. It was most likely a wishful thinking to have it work out of the box! :) My current suspicion is that the `section` may not be supported as subsection of output?! Anyway, since this is fairly secondary, I think we should leave it for now, the question is do we want to have it merged or removed from this PR?

**Comment by douglowe** [comment]
Yep, there do seem to be (small, subtle) differences in the way that subpackages treat sections compared to using them in the main package.

I think it might be best to leave the 'app' section out of this PR - and make it it's own PR, so we can concentrate there just on trying to get it to work.

**Comment by ericpre** [comment]
> I think it might be best to leave the 'app' section out of this PR - and make it it's own PR, so we can concentrate there just on trying to get it to work.

Done.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 processor:
   - cpu
-
+vc:
+- 14
```
Comment (douglowe) [inline recipe/conda_build_config.yaml]: What's vc? Also, do there need to be spaces before the `-14` (similar to the `  - cpu` line)?

**Review by douglowe — APPROVED** [review]
It looks okay to me - and I'm happy that the OSX setup still works fine. So I think (pending a check on my comment for the conda_build_config.yaml file) that it can be merged.

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (ericpre) [reply recipe/conda_build_config.yaml]: Indeed, this should be removed, I needed it to pick up the right compiler on windows when building locally and I have committed it by mistake; it is removed now. Normally, it should make no difference on azure pipelines.

Comment (douglowe) [reply recipe/conda_build_config.yaml]: cool. I guess we could merge it now then.

**Comment by ericpre** [comment]
Thanks @douglowe for the review!

**Pull request merged**